src/components/link/Link.tsx

```diff
@@ -19,9 +19,9 @@ interface LinkProps extends Omit<BoxProps, 'ref'> {
   /** A custom element to be rendered */
   element?: Element | string;
   /** Callback function that is fired when mouse leaves the component. */
-  onMouseLeave?: () => void;
+  onMouseLeave?: (event: MouseEvent) => void;
```

Comment:
To be 100% correct this should be React.MouseEvent (or import MouseEvent from react) otherwise the type is the [native mouse event](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent)

And technically adding `onMouseLeave` and `onMouseUp` (+children + className + element) here shouldn't be required as they should already be  present on the `BoxProps` type